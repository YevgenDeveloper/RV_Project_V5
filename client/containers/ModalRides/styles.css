@import 'client/styles/variables';

.header {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 20px;

  h1 {
    margin-bottom: 0;
  }

  .buttonAdd {
    width: 100px;
    height: 32px;
    overflow: hidden;
    position: relative;
    margin-left: auto;

    .button {
      z-index: 1;
      pointer-events: none;
    }

    input {
      z-index: 2;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      position: absolute;
      opacity: 0;
    }
  }
}

.rides {
  display: flex;
  flex-flow: row wrap;
}

.ride {
  width: 100%;
  padding: 15px 0;
  display: block;
  text-align: left;
  cursor: pointer;
  border-top: 1px solid rgba(0, 0, 0, 0.1);

  &:last-child {
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }
}
