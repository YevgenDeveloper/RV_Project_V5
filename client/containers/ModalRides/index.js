// @flow

import React from 'react';
import classnames from 'classnames/bind';

import { Modal, Button } from 'client/components';
import styles from './styles.css';

const cx = classnames.bind(styles);

type Props = {
  rides: string[],
  currentRide: string,
  close: void => void,
  onChangeRide: (ride: string) => void,
};

const ModalRides = (props: Props) => (
  <Modal close={props.close}>
    <div className={styles.header}>
      <h1>Rides</h1>
      <div className={styles.buttonAdd}>
        <Button
          blue
          label="Add ride"
          className={styles.button}
          onClick={() => {}}
        />
        <input type="file" />
      </div>
    </div>
    <div className={styles.rides}>
      {props.rides.map(ride => (
        <button
          key={ride}
          type="button"
          className={cx(styles.ride, {
            [styles.on]: ride === props.currentRide,
          })}
          onClick={() => props.onChangeRide(ride)}
        >
          {ride}
        </button>
      ))}
    </div>
  </Modal>
);

export default ModalRides;
