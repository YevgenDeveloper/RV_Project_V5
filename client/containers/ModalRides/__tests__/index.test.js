// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import ModalRides from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <ModalRides
      rides={[]}
      currentRide="Test"
      onChangeRide={jest.fn()}
      close={jest.fn()}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
