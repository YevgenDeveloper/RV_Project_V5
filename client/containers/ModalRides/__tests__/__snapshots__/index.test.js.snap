// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalRides
    close={[MockFunction]}
    currentRide="Test"
    onChangeRide={[MockFunction]}
    rides={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="header"
        >
          <h1>
            Rides
          </h1>
          <div
            className="buttonAdd"
          >
            <Button
              blue={true}
              className="button"
              dark={false}
              disabled={false}
              label="Add ride"
              onClick={[Function]}
            />
            <input
              type="file"
            />
          </div>
        </div>,
        <div
          className="rides"
        />,
      ],
      "className": undefined,
      "close": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Rides
            </h1>,
            <div
              className="buttonAdd"
            >
              <Button
                blue={true}
                className="button"
                dark={false}
                disabled={false}
                label="Add ride"
                onClick={[Function]}
              />
              <input
                type="file"
              />
            </div>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rides",
            },
            "ref": null,
            "rendered": "Rides",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  blue={true}
                  className="button"
                  dark={false}
                  disabled={false}
                  label="Add ride"
                  onClick={[Function]}
                />,
                <input
                  type="file"
                />,
              ],
              "className": "buttonAdd",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "blue": true,
                  "className": "button",
                  "dark": false,
                  "disabled": false,
                  "label": "Add ride",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "rides",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="header"
          >
            <h1>
              Rides
            </h1>
            <div
              className="buttonAdd"
            >
              <Button
                blue={true}
                className="button"
                dark={false}
                disabled={false}
                label="Add ride"
                onClick={[Function]}
              />
              <input
                type="file"
              />
            </div>
          </div>,
          <div
            className="rides"
          />,
        ],
        "className": undefined,
        "close": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Rides
              </h1>,
              <div
                className="buttonAdd"
              >
                <Button
                  blue={true}
                  className="button"
                  dark={false}
                  disabled={false}
                  label="Add ride"
                  onClick={[Function]}
                />
                <input
                  type="file"
                />
              </div>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rides",
              },
              "ref": null,
              "rendered": "Rides",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    blue={true}
                    className="button"
                    dark={false}
                    disabled={false}
                    label="Add ride"
                    onClick={[Function]}
                  />,
                  <input
                    type="file"
                  />,
                ],
                "className": "buttonAdd",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "blue": true,
                    "className": "button",
                    "dark": false,
                    "disabled": false,
                    "label": "Add ride",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "type": "file",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "rides",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
